src/Quizzer.API.BL/Facades/QuestionFacade.cs

```diff
@@ -0,0 +1,308 @@
+            using IRepository<QuestionEntity> repository = uow.GetRepository<QuestionEntity>();
+
+            IEnumerable<QuestionEntity> questionEntities = repository.GetByPredicate(q => q.QuizId == quizId, nameof(QuestionEntity.Answers),
+                nameof(QuestionEntity.UserAnswers), $"{nameof(QuestionEntity.UserAnswers)}.{nameof(UserAnswerEntity.SelectedAnswers)}");
+
+            var questionModels = new List<QuestionResultModel>();
+            foreach (QuestionEntity questionEntity in questionEntities) {
+
+                QuestionResultModel quesitonModel = p_mapper.Map<QuestionResultModel>(questionEntity);
+
+                List<AnswerResultModel> answerModels = MapAnswers(userId, questionEntity);
+                quesitonModel.Answers = answerModels;
+
+                quesitonModel.IsAnswered = answerModels.Any(a => a.IsAnswered);
+
+                questionModels.Add(quesitonModel);
+            }
+
+            return questionModels;
+        }
+
+        public override Guid Save(QuestionDetailModel model) {
+            if (string.IsNullOrEmpty(model.Text)) throw new ArgumentException("Question text must not be empty");
+
+            IUnitOfWork uow = p_unitOfWorkFactory.Create();
+
+            using (IRepository<QuestionEntity> repository = uow.GetRepository<QuestionEntity>()) {
+                QuestionEntity? entity = repository.GetById(model.Id);
+
+                if (entity is not null) HandleFormatChange(entity, model.QuestionType);
+            }
+
+            model.Text = model.Text.Trim();
+
+            return base.Save(model);
+        }
+
+        public void AddQuestionToQuiz(Guid questionId, Guid quizId) {
+            IUnitOfWork uow = p_unitOfWorkFactory.Create();
+            using (IRepository<QuestionEntity> questionRepository = uow.GetRepository<QuestionEntity>()) {
+                using (IRepository<QuizEntity> quizRepository = uow.GetRepository<QuizEntity>()) {
+                    if (!quizRepository.Exists(quizId)) throw new ArgumentException("Quiz does not exist");
+                }
+
+                QuestionEntity? entity = questionRepository.GetById(questionId)
+                    ?? throw new ArgumentException("Question does not exist");
+
+                entity.QuizId = quizId;
+
+                questionRepository.Update(entity);
+            }
+            uow.Commit();
+        }
+
+        public void AnswerQuestion(Guid answerId, Guid userId, DateTime answerTime, int order = 0) {
+            IUnitOfWork uow = p_unitOfWorkFactory.Create();
+
+            using IRepository<SelectedAnswerEntity> selectedAnswerRepository = uow.GetRepository<SelectedAnswerEntity>();
+
+            AnswerEntity answerEntity = GetValidAnswer(answerId, uow);
+
+            CheckQuizTimeFrame(answerTime, answerEntity);
+
+            UserAnswerEntity userAnswer = GetUserAnswer(userId, answerTime, uow, answerEntity.QuestionId);
+
+            switch (answerEntity?.Question?.Type ?? AnswerFormat.Undefined) {
```

Comment:
@KoubaFromPucks The reason why I introduced the undefined Answer format is this switch, which must be null safe, I am not sure how switch will handle null value (probably it will pass it to the default as the compilation did not raise any errors). 
In terms of `GetValidAnswer` method, it should return a mapped question, but it is uncertain whether this is 100% the case, but I tried to ensure it by adding question ID in AnswerDetailModel and checking the presence in `AnswerFacade`:
 https://github.com/vecerap/iw5-2023/blob/b318110cf4918bee83ef5252a74189cd45150e61/src/Quizzer.API.BL/Facades/AnswerFacade.cs#L51

But as for now, I will delete this, because apparently [the switch statement points null values to the default label, unless specified different.](https://stackoverflow.com/questions/14950532/how-will-a-c-sharp-switch-statements-default-label-handle-a-nullable-enum)

My tests proves those results even with numbers:
![image](https://github.com/vecerap/iw5-2023/assets/93704213/47378c07-65d2-46a7-8142-0a3f84402a09)

Reply:
ok